### DISCUSSION TITLE
Get typesafety for events ?

### DISCUSSION THREAD

**Discussion by krisgrm** [wevm/wagmi#2003] [Q&A]
Hey using a contract read from generated contract (or generally even from typechain).. when using

```
contract.on('eventName', ()... {
}
```

eventName is not typesafe :(

**Comment by tmm** [comment]
Check out [`getContract`](https://wagmi.sh/core/actions/getContract) and the [wagmi CLI](https://wagmi.sh/cli/getting-started).
